src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -101,39 +92,55 @@ public void RefreshErrorList()
         {
             // Mark all the sinks as dirty (so, as a side-effect, they will start an update pass that will get the new snapshot
             // from the snapshot factories).
-            lock (managers)
+            lock (sinks)
             {
-                foreach (var manager in managers)
+                foreach (var sink in sinks)
                 {
-                    manager.UpdateSink();
+                    SafeOperation(sink, "FactorySnapshotChanged", () => sink.FactorySnapshotChanged(null));
```

Comment:
We added this error handling in the past to work round a problem with a bad sink (see #1055).
The error handling wasn't previously in the `Subscribe` or `Dispose` methods so I haven't added it to them.